db/api/src/settings.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{db::HoprDb, errors::Result};
+use async_trait::async_trait;
+use hopr_internal_types::prelude::AliasEntry;
+use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, QueryFilter};
+
+pub const ALIASES_ENTRY_NAME: &str = "aliases";
+
+/// Defines DB API for accessing HOPR settings (mainly aliases for now)
+#[async_trait]
+pub trait HoprDbSettingsOperations {
+    /// Retrieve the alias for a given PeerID
+    async fn get_alias(&self, alias: String) -> Result<Option<String>>;
+
+    /// Retrive all aliases
+    async fn get_aliases(&self) -> Result<Vec<AliasEntry>>;
+
+    /// Create new key pair value in the db
+    async fn set_alias(&self, peer: String, alias: String) -> Result<bool>;
+
+    /// Update aliases
+    async fn set_aliases(&self, aliases: Vec<AliasEntry>) -> Result<bool>;
+
+    /// Delete alias
+    async fn delete_alias(&self, alias: String) -> Result<bool>;
+}
+
+#[async_trait]
+impl HoprDbSettingsOperations for HoprDb {
+    async fn get_alias(&self, alias: String) -> Result<Option<String>> {
+        let aliases = self.get_aliases().await?;
+        let alias_entry = aliases.into_iter().find(|entry| entry.alias == alias);
+
+        match alias_entry {
+            Some(alias_entry) => return Ok(Some(alias_entry.peer_id)),
+            None => return Ok(None),
+        }
+    }
+
+    async fn get_aliases(&self) -> Result<Vec<AliasEntry>> {
+        let model = hopr_db_entity::global_settings::Entity::find()
+            .filter(hopr_db_entity::global_settings::Column::Key.eq("aliases"))
+            .one(&self.settings_db)
+            .await?;
+
+        if model.is_none() {
+            let model = hopr_db_entity::global_settings::ActiveModel {
+                id: Default::default(),
+                key: sea_orm::ActiveValue::Set(ALIASES_ENTRY_NAME.to_string()),
+                value: sea_orm::ActiveValue::Set("{}".into()),
+            };
+
+            model.insert(&self.settings_db).await?;
+            return Ok(vec![]);
+        }
+
+        let model = model.unwrap();
+        let aliases_row = std::str::from_utf8(&model.value).unwrap();
+        let aliases: Vec<AliasEntry> = serde_json::from_str(aliases_row).unwrap_or_default();
```

Comment:
Unhandled exceptions directly in the code, process abort. Never use `unwrap`, when you can fail.